app/src/main/java/net/squanchy/remoteconfig/RemoteConfig.java

```diff
@@ -25,11 +26,17 @@
     }
 
     public Single<Boolean> proximityServicesEnabled() {
-        return getBooleanConfigValue(() -> remoteConfig.getBoolean(KEY_PROXIMITY_ENABLED))
+        return getConfigValue(() -> remoteConfig.getBoolean(KEY_PROXIMITY_ENABLED))
                 .subscribeOn(Schedulers.io());
     }
 
-    private <T> Single<T> getBooleanConfigValue(Func0<T> action) {
+    public Single<Integer> contestGoal() {
+        return getConfigValue(() -> remoteConfig.getLong(KEY_CONTEST_GOAL))
+                .map(Long::intValue)
```

Comment:
This is generally not safe (might be in this case, or might not if there's a typo), I'd rather carry it around as a `long`